staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/grpc_service.go

```diff
@@ -138,3 +142,13 @@ func (g *gRPCService) Status(ctx context.Context) (*kmsservice.StatusResponse, e
 	}
 	return &kmsservice.StatusResponse{Version: response.Version, Healthz: response.Healthz, KeyID: response.KeyId}, nil
 }
+
+func (g *gRPCService) recordMetricsInterceptor() grpc.UnaryClientInterceptor {
```

Comment:
We can avoid the indirection through the struct:

```suggestion
func recordMetricsInterceptor(providerName string) grpc.UnaryClientInterceptor {
```